astropy/coordinates/tests/test_funcs.py

```diff
@@ -33,6 +33,38 @@ def test_sun():
     gcrs2 = get_sun(Time([northern_summer_solstice, equinox_2, northern_winter_solstice]))
     assert np.all(np.abs(gcrs2.dec - [23.5, 0, -23.5]*u.deg) < 1*u.deg)
 
+def test_sun_02():
+    """
+    Test `that get_sun` produces similar values to skyfield (de421).
+
+    Commands to produce skyfield values:
+
+    from skyfield.api import sun, earth
+    from skyfield.units import Angle
+
+    apparent_gcrs = earth(utc=(2010,6,21)).observe(sun).apparent().radec()
+
+    skyf_ra_apparent = Angle(apparent_gcrs[0]).degrees
+    skyf_dec_apparent = Angle(apparent_gcrs[1]).degrees
+    skyf_dist_apparent = apparent_gcrs[2].AU
+    """
+    from ..funcs import get_sun
+
+    time_apy = Time('2010-6-21')
+    gcrs_apy = get_sun(time_apy)
+
+    apy_ra = gcrs_apy.ra.deg
+    apy_dec = gcrs_apy.dec.deg
+    apy_dist = gcrs_apy.distance.AU
+    
+    skyf_ra_apparent = 89.338458132829359
+    skyf_dec_apparent = 23.436389712068134 
+    skyf_dist_apparent = 1.016198586488303
+
+    assert abs(apy_ra - skyf_ra_apparent) < 0.01
```

Comment:
In this case, an alternative is to use astropy quantity comparisons when possible so that you don't ever have to worry about the units.  E.g.:
```
assert abs(apy_ra.ra gcrs_apy.ra - skyf_ra_apparent*u.deg < 30*u.arcsec)
```
and if you do that you don't need the ``apy_ra = gcrs_apy.ra.deg`` line at all.  

But as @cdeil mentioned, `assert_allclose` makes numeric comparisons often easier anyway.  So the way to do that without the extra ``.deg`` would be something like this:
```
assert_allclose(gcrs_apy.ra.deg, skyf_ra_apparent, atol=(30*u.arcsec).to(u.deg).value)
```
(or ``atol=0.01`` to get just what you have here)